### PR TITLE
feat(ctb): Add doOwnershipTransfer to deploy utils

### PR DESCRIPTION
**Pull request by maurelian**
1. Creates a reusable function `doOwnershipTransfer` which deduplicates the logic for transferring ownership of the 4 system contracts to the MSD.
2. Combines the `awaitCondition` call for each of those transfers into a single call at the end. This provides better support for a `multicall` based transfer, by printing out all the info required to create a batch of transactions at once. It also still supports a 'one at a time' transfer approach, as it will check all contracts before progressing.

### CODE REVIEW

**Comment by maurelian** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5352** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5352" target="_blank">Graphite</a> 
    * **PR #5353** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5353" target="_blank">Graphite</a> 
      * **PR #5354** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5354" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/5354?utm_source=stack-comment).

File: packages/contracts-bedrock/src/deploy-utils.ts [hunk 1]
```diff
@@ -322,6 +322,39 @@ export const printJsonTransaction = (tx: ethers.PopulatedTransaction): void => {
...
+ *
+ * @param opts Options for executing the step.
+ * @param opts.isLiveDeployer True if the deployer is live.
+ * @param opts.proxy proxy contract.
+ * @param opts.dictator dictator contract.
+ */
+export const doOwnershipTransfer = async (opts: {
+  isLiveDeployer?: boolean
+  proxy: ethers.Contract
+  name: string
```
Comment (maurelian) [inline packages/contracts-bedrock/src/deploy-utils.ts:336]: I don't like passing the name as an arg, I'm sure there's a better way, but good enough for now IMO.

File: packages/contracts-bedrock/deploy/020-SystemDictatorSteps-1.ts [hunk 2]
```diff
@@ -73,10 +71,13 @@ const deployFn: DeployFunction = async (hre) => {
...
-    await ProxyAdmin.transferOwnership(SystemDictator.address)
-  } else {
-    console.log(`Proxy admin already owned by MSD`)
+    await doOwnershipTransfer({
+      isLiveDeployer,
+      proxy: ProxyAdmin,
+      name: 'ProxyAdmin',
+      transferFunc: 'transferOwnership',
+      dictator: SystemDictator,
+    })
```
Comment (maurelian) [inline packages/contracts-bedrock/deploy/020-SystemDictatorSteps-1.ts:80]: I recognize that the proxy admin is always owned by the `deployer` key as it is configured that way in the deployment scripts, so this doesn't use all the functionality of the new function, but I still think its cleaner.

**Review by tynes — APPROVED** [review]
This is a great addition, good work

**Pull request merged**